src/brsTypes/components/RoAssociativeArray.ts

```diff
@@ -182,7 +228,9 @@ export class RoAssociativeArray extends BrsComponent implements BrsValue, BrsIte
                 return BrsInvalid.Instance;
             }
 
-            this.elements = new Map<string, BrsType>([...this.elements, ...obj.elements]);
+            obj.elements.forEach((value, key) => {
+                this.set(new BrsString(key), value, true);
+            });
```

Comment:
what is this logic change needed for?

Reply:
cuz with a new approach, we need add lowercase key to map too, and save original key, what `this.set` already do